sources/projects/base64/description.md

```diff
@@ -0,0 +1,22 @@
+This program can encode text to base64-encoded text, and it can decode
+base64-encoded text into normal text.
+
+Base64 is a popular method of encoding strings and other data. It can encode
+images, text, JSON, and almost any other format as well.
+
+It is also often URL-safe.
+
+Base64 works by splitting the data into 6-bit segments, which are then mapped
+to a character in the table.
+
+For example, let's use the string `six`.
+
+This would be split into `1110011 1101001 1111000`. However, these are 8-bit
+groups. Base64 needs 6-bit groups. So this turns into `111001 111010 011111
+000` (in decimal: `57 58 31`).
+
+Then, we take an index from a character table. Character 57 is `c`, so we add that.
+
+Once we've added all the characters, we're done!
```

Comment:
Not quit:

The input is divided into 24-bit (3 byte) chunks. Each 24-bit is encoded as 4 ASCII characters from the table. There is also the case where the length of the input is not a multiple of three. In this case, pad characters (`=`) are added so that the output is a multiple of four.

You'll need to show what happens when there is one and two leftover characters. 

It would probably be a good idea to show the table. You'll need to describe the decode logic.